### PR TITLE
Testing Tic Tac Toe

### PR DESCRIPTION
**Pull request by bruna-genz**
In this brach we created unit tests for our modules using Jest.

Ps.: we did not create a test file for the displayModule file because all functions from this module interacts with the UI.

### CODE REVIEW

File: src/js/tests/board.test.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+test('newboard function should return a board an empty array', () => {
+  expect(boardGame.newBoard()).toEqual(boardArray);
+});
+
+test("update mark X on index 2 array's game", () => {
+  boardGame.newBoard();
+  boardGame.updateArray(2, 'X');
+  expect(boardGame.getBoard()).toEqual([' ', ' ', 'X', ' ', ' ', ' ', ' ', ' ', ' ']);
+});
```
Comment (bolah2009) [inline src/js/tests/board.test.js]: For the `boardGame` test, let's test for each public methods (`newBoard`, `getBoard`, `updateArray`), these methods should have their own test examples each.

Also, remember about negative and positive cases.

File: src/js/tests/gameModule.test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+describe('when new game starts', () => {
+  test("status should be 'on'", () => {
+    playerFactory.mockReturnValueOnce(mockPlayer0);
+    playerFactory.mockReturnValueOnce(mockPlayer1);
+
+    gameModule.newGame();
+
+    expect(gameModule.getStatus()).toBe('on');
+  });
```
Comment (bolah2009) [inline src/js/tests/gameModule.test.js]: Like the previous comment, let's cover tests for all the methods of the `gameModule` public methods too (`gameInit`, `newGame`, `makeMove`, `getStatus`, `getCurrentPlayer`, `cleanBoard`, `setNames`, `getPlayer0`, `getPlayer1`).

**Review by bolah2009 — CHANGES_REQUESTED** [review]
Awesome job with the tests!

Let's test all the public methods of each function used, see comments below for more details.

#### :recycle:  Make these changes and submit another code review request. 👍🏾

#### Feel free to reach me on slack if you having issues implementing these features. :relaxed: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

**Comment by bruna-genz** [comment]
We made more tests as suggested.

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,
Thank you for the improvements. However, there is a bug to fix.

1. I got this error when running `npm run test`
```js
$ npm run test

> Tic-Tac-Toe@1.0.0 test /home/maelfosso/Documents/Microverse/TSE/Js/Tic-Tac-Toe
> jest --watch .

internal/fs/watchers.js:173
    throw error;
    ^

Error: ENOSPC: System limit for number of file watchers reached, watch '/home/maelfosso/Documents/Microverse/TSE/Js/Tic-Tac-Toe/node_modules/body-parser/node_modules/qs/dist'
    at FSWatcher.start (internal/fs/watchers.js:165:26)
    at Object.watch (fs.js:1329:11)
    at NodeWatcher.watchdir (/home/maelfosso/Documents/Microverse/TSE/Js/Tic-Tac-Toe/node_modules/sane/src/node_watcher.js:159:22)
    at Walker.<anonymous> (/home/maelfosso/Documents/Microverse/TSE/Js/Tic-Tac-Toe/node_modules/sane/src/common.js:109:31)
    at Walker.emit (events.js:223:5)
    at /home/maelfosso/Documents/Microverse/TSE/Js/Tic-Tac-Toe/node_modules/walker/lib/walker.js:69:16
    at FSReqCallback.oncomplete (fs.js:146:23) {
  errno: -28,
  syscall: 'watch',
  code: 'ENOSPC',
  path: '/home/maelfosso/Documents/Microverse/TSE/Js/Tic-Tac-Toe/node_modules/body-parser/node_modules/qs/dist',
  filename: '/home/maelfosso/Documents/Microverse/TSE/Js/Tic-Tac-Toe/node_modules/body-parser/node_modules/qs/dist'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! Tic-Tac-Toe@1.0.0 test: `jest --watch .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the Tic-Tac-Toe@1.0.0 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/maelfosso/.npm/_logs/2020-05-24T06_51_47_983Z-debug.log
```

2. I suggest you move the `src/js/tests` folder at the root, the same level than `src`
Otherwise, testing the `.test.js` files one by one is working perfectly!

3. Update your README with informations concerning `jest`

Kindly make the changes and submit a new code review request
Thanks

Mael FOSSO

**Review by fatymahmed — APPROVED** [review]
## Hi @bruna-genz @nicolasmarino99,
### Great job making the required changes 🎉🎉
### It's a 👍🏾 from me!
## ✔️ Approved!

**Pull request merged**